src/python/pants/testutil/pants_run_integration_test.py

```diff
@@ -557,42 +558,83 @@ def file_renamed(self, prefix, test_name, real_name):
             os.rename(real_path, test_path)
 
     @contextmanager
-    def temporary_file_content(self, path, content, binary_mode=True):
+    def temporary_directory_literal(
+        self, path: Union[str, Path],
+    ):
+        """Temporarily create the given literal directory, which must not exist."""
+        path = os.path.realpath(path)
+        assert path.startswith(
```

Comment:
This assert is surprising given the doc. Seems worth documenting or else moving the caring about this to the call site that has this restriction.